python/pyspark/errors/utils.py

```diff
@@ -172,19 +181,38 @@ def _with_origin(func: Callable[..., Any]) -> Callable[..., Any]:
     @functools.wraps(func)
     def wrapper(*args: Any, **kwargs: Any) -> Any:
         from pyspark.sql import SparkSession
+        from pyspark.sql.utils import is_remote
 
         spark = SparkSession.getActiveSession()
         if spark is not None and hasattr(func, "__name__"):
-            assert spark._jvm is not None
-            pyspark_origin = spark._jvm.org.apache.spark.sql.catalyst.trees.PySparkCurrentOrigin
+            if is_remote():
+                global current_origin
 
-            # Update call site when the function is called
-            _capture_call_site(spark, pyspark_origin, func.__name__)
+                # Getting the configuration requires RPC call. Uses the default value for now.
+                depth = 1
```

Comment:
what's the purpose of depth?

Reply:
It shows the maximum traceback of the error. For example, when it's `1`:

```
...
== DataFrame ==
"divide" was called from
/.../spark/python/test_pyspark_error.py:4
```

when it's 2:

```
...
== DataFrame ==
"divide" was called from
/.../spark/python/test_pyspark_error.py:4
/.../spark/python/test_parent_pyspark_error.py:5
```